### ISSUE TITLE
Metric for Node Launch/Startup time

### ISSUE THREAD

**Issue by dumlutimuralp** [aws/karpenter-provider-aws#7171]
### Description

**What problem are you trying to solve?**

Organizations would like to have a metric that shows the time it takes for a node to boot and get into ready state in Kubernetes. Currently Karpenter Prometheus metrics show [karpenter_pods_startup_duration_second](https://karpenter.sh/docs/reference/metrics/#karpenter_pods_startup_duration_seconds) only.

**How important is this feature to you?**

Having highly dynamic workloads and multi tenant environments node startup time becomes an important metric that impacts multiple dimensions. In scaling events it is crucial for organizations to be able to measure the overall scaling behavior based on this metric. 

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

**Comment by njtran** [comment]
This should be captured by the metrics for the NodeClaim Initialization status condition. You can get different details around when this status condition is added here: https://karpenter.sh/docs/reference/metrics/#status-condition-metrics


TLDR is that a node is ready for pod scheduling and use once the NodeClaim for a Karpenter node has the status condition Initialized = true.
More info on what Launched/Registered/Initialized is here https://karpenter.sh/docs/concepts/nodeclaims/

**Comment by youwalther65** [comment]
@dumlutimuralp It seems one can use the metric `operator_status_condition_current_status_seconds`:

`operator_status_condition_current_status_seconds{kind="NodeClaim", name="<node claim name>",status="True", type="Launched"} - ignoring(type,reason) operator_status_condition_current_status_seconds{kind="NodeClaim", name="<node claim name>",status="True", type="Ready"}`

 Comparing this number, which is seconds from `Launched` to `Ready` with Karpenter logs or nodeclaim.status section it seems it really shows the right values.

**Comment by andy-townsend** [comment]
It looks like `operator_status_condition_current_status_seconds` is deprecated in v1.1 and I've also noticed some odd results with a similar query to the above, as in its suggesting some nodes take 3-4 days to start
